.github/workflows/publish.yaml

```diff
@@ -25,14 +25,13 @@ jobs:
       - name: Run tests
         run: flutter test
       - name: Create credentials directory
-        run: mkdir ~/.pub-cache
+        run: mkdir ~/.config/dart
       - name: Setup Pub Credentials
         shell: bash
         env:
-          DART_PUBLISH_ACCESS_TOKEN: ${{ secrets.DART_PUBLISH_ACCESS_TOKEN }}
-          DART_PUBLISH_REFRESH_TOKEN: ${{ secrets.DART_PUBLISH_REFRESH_TOKEN }}
+          DART_PUB_CREDENTIALS_JSON: ${{ secrets.DART_PUB_CREDENTIALS_JSON }}
         run: |
-          sh ./scripts/pub_login.sh
+          cat "$DART_PUB_CREDENTIALS_JSON" > ~/.config/dart/pub-credentials.json
```

Comment:
Because it's in the raw runtime - you no longer need the `env` passthrough - you can access the raw secret here.